sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/models/FeatureFlagConfigurationSetting.java

```diff
@@ -40,12 +76,64 @@ public final class FeatureFlagConfigurationSetting extends ConfigurationSetting
      * @param isEnabled A boolean value to turn on/off the feature flag setting.
      */
     public FeatureFlagConfigurationSetting(String featureId, boolean isEnabled) {
+        isValidValue = true;
+
         this.featureId = featureId;
         this.isEnabled = isEnabled;
         super.setKey(KEY_PREFIX + featureId);
         super.setContentType(FEATURE_FLAG_CONTENT_TYPE);
     }
 
+    @Override
+    public String getValue() {
+        // Lazily update: Update 'value' by all latest property values when this getValue() method is called.
+
+        // If the 'value' wasn't valid, return the original 'value'.
+        if (!isValidValue) {
```

Comment:
Does this mean it isn't valid JSON? The value of a feature flag should always be valid JSON, as it's part of the content-type.

Reply:
FeatureFlagConfigurationSetting is also a ConfigurationSetting. So when user calls getValue() it should return a value.

Reply:
My question is what is an invalid Value?

Reply:
An invalid featureflag value. Sorry for the confuse.

Reply:
That invalid value shouldn't be allowed to be set. It will fail if sent to app configuration.

Reply:
If you remove the code and try to parse it as is, then it will result in an error which would be correct.

Or you could make value not settable, and make it so we have a "other" map which updates the value field. I think another language does that.

Reply:
Hmm. I tried to align with what .NET shipped: https://github.com/Azure/azure-sdk-for-net/blob/f41d1a08ff2c2b8d748c61774caac5491bf18b88/sdk/appconfiguration/Azure.Data.AppConfiguration/src/Models/FeatureFlagConfigurationSetting.cs#L192

Reply:
After an offline discussion with Matt. Java will throw an exception if the value is invalid instead of returning the original value.
It aligns with the preview version behavior in Java.